include/maths/analytics/CBoostedTreeLeafNodeStatistics.h

```diff
@@ -459,12 +484,26 @@ class MATHS_ANALYTICS_EXPORT CBoostedTreeLeafNodeStatistics {
         using TDerivatives2x1 = Eigen::Matrix<double, 2, 1>;
 
     private:
-        static std::size_t number(const TDerivativesVec& derivatives) {
-            return derivatives.size();
+        void initializeAndMap(const TFloatVecVec& splits) {
```

Comment:
Well in this case I would argue that they do do to separate things: `initialiseAndMaps` zeros memory then maps the addresses, and `copyAndMap` copies the memory from the other derivatives and then maps the addresses. So I think I would prefer to keep different names. I renamed though to mention storage since this now only manages part of the object state.